macros/relationships/append_only/aggregate_in_before.sql

```diff
@@ -0,0 +1,23 @@
+{% macro aggregate_in_before_join_clause(i) %}
+
+{% set stream = dbt_activity_schema.generate_stream_alias %}
+{% set columns = dbt_activity_schema.columns() %}
+
+(
+    {{ stream(i) }}.{{- columns.ts }} < {{ stream() }}.{{- columns.ts }}
+)
+{% endmacro %}
+
+{% macro aggregate_in_before_aggregation_func() %}
+count(distinct {{ caller() }})
+{% endmacro %}
+
+{% macro aggregate_in_before() %}
```

Comment:
I was thinking we could actually allow users to pass their own aggregation macros here:

```suggestion
{% macro aggregate_in_before(agg_func=dbt_activity_schema.aggregate_in_before_aggregation_func) %}
```